app/assets/javascripts/host_edit.js

```diff
@@ -591,6 +591,27 @@ $(document).on('change', '.interface_type', function () {
   interface_type_selected(this);
 });
 
+$(document).on('change', '.host_name', function (event) {
```

Comment:
It seems Jenkins also hits into some issues in relation to this code http://ci.theforeman.org/job/test_develop_pr_core/13520/